src/RustyObjectStore.jl

```diff
@@ -81,52 +98,193 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
     return nothing
 end
 
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `request_timeout_secs::Int`: (Optional) Client request timeout in seconds.
+- `connect_timeout_secs::Int`: (Optional) Client connection timeout in seconds.
+- `max_retries::Int`: (Optional) Maximum number of retry attempts.
+- `retry_timeout_secs::Int`: (Optional) Maximum amount of time from the initial request after which no further retries will be attempted (in seconds).
+"""
+struct ClientOptions
+    params::Dict{String, String}
+
+    function ClientOptions(;
+        request_timeout_secs::Int = 30,
```

Comment:
Can we make these options and use the default on the rust side if they aren't set?

Reply:
Will do!